### DISCUSSION TITLE
Standardizing the inputs for search filters

### DISCUSSION THREAD

**Discussion by gargmishu** [Sunbird-Knowlg/Community#26] [Feature Requests]
I'm exploring Diksha APIs on sandbox https://dev.diksha.gov.in/#/discoverapi and noticed that lot of filters are free text string.

The filters for content search API (/content/v1/search) takes free text as input, which is subject to typos. API users might give invalid input inadvertently and end up with no results. There is no way to know if result set is actually empty or due to invalid inputs. This can be prevented by standardizing inputs

ex: se_grades could take only numerical inputs from class 1-10 and throw error if someone specifies any other value
se_subject, se_medium, primaryCategory - could take predefined values (similar to enums in java) and throw error if any other input is specified.

Happy to engage in discussion and discuss more about the idea

**Comment by vinukumar-vs** [comment]
@gargmishu Thanks for the suggestion. 
We have to look at ElasticSearch capability to handle this. Lets explore will ES give this capability out of the box instead we build our custom logic.

@AmiableAnil @Krishnaj20

**Reply by gargmishu** [reply]
thanks for consideration. Happy to engage in discussions if you prefer.
